### ISSUE TITLE
Why a reduced statistic smaller than 3 is required to estimate errors with the confidence method? 

### ISSUE THREAD

**Issue by cosimoNigro** [sherpa/sherpa#1196]
Hi everyone,

I am fitting a radiative model to a MWL SED, the final statistics I obtain (`chi2`) is not optimal
```
fit succesful? True
Method                = levmar
Statistic             = chi2
Initial fit statistic = 462.681
Final fit statistic   = 435.362 at function evaluation 51
Data points           = 86
Degrees of freedom    = 80
Probability [Q-value] = 2.58777e-50
Reduced statistic     = 5.44203
Change in statistic   = 27.3188
```
after this, if I try to estimate the errors via `fitter.est_errors()`, having set `fitter.estmethod = Confidence()`, I obtain the following error
```
Traceback (most recent call last):
  File "figure_6_Mrk421_fit_sherpa.py", line 245, in <module>
    errors_2 = fitter.est_errors()
  File "/home/cosimo/software/miniconda3/envs/gammapy-0.18.2/lib/python3.7/site-packages/sherpa/fit.py", line 54, in run
    result = func(fit, *args, **kwargs)
  File "/home/cosimo/software/miniconda3/envs/gammapy-0.18.2/lib/python3.7/site-packages/sherpa/fit.py", line 1440, in est_errors
    raise EstErr('rstat>max', str(self.estmethod.max_rstat))
sherpa.utils.err.EstErr: reduced statistic larger than 3
```
I apologise for my ignorance but:
1) is there a statistical reason why a final reduced statistics larger than 3 prevents the computation of a confidence interval? Or there is some numerical reason behind it?
2) is there some option in `est_errors` that can be set to increase this limit? In the [docs](https://sherpa.readthedocs.io/en/latest/fit/api/sherpa.fit.Fit.html?highlight=est_errors#sherpa.fit.Fit.est_errors) a `methoddict` option is specified, but I think it has to do with re-optimisation. I think in my case the calculation is not even starting due to the limit on `chi2 / d.o.f.`.

Thanks a lot!

**Comment by dtnguyen2** [comment]
The answer to the first question: The following excerpt:
```
The error estimate should only be run when at a “good” fit. The assumption is that the search statistic is chi-square distributed so the change in statistic as a statistic varies can be related to a confidence bound. One requirement is that - for chi-square statistics - is that the reduced statistic value is small enough. See the max_rstat field of the EstMethod object.

Give the error if this does not happen (e.g. if c1 has not been fit)?
```
was taken from [override using max_rstat](https://sherpa.readthedocs.io/en/latest/fit/index.html)

As for the second question, override it with ```max_rstat```

**Comment by hamogu** [comment]
I believe that the functions would in principle run with a Chi^2 > 3, in the sense that they would give you a number out. However, that number is only meaningful if your model is good, because the calculations implicitly assume that the model is nicely behaved, e.g. the Chi^2 distributions is Gaussian around the minimum Chi^2 value. For bad models (those that don't fit well and have large Chi^2 values) that is generally not the case. So, the check if chi^2 > 3 is there to protect the user from getting numbers which are wrong. That's why there is little use in changing the "3" to some other number. (There is no statistical reason why our cut-off is exactly at 3.0 and not e.g. 2.5 or 4.0 though, but the 5.44 that you get for your data is definitely too large.)

So, the question is why your Chi^2 is so large. There are several possible explanations, but since I don't know your data and your model, I don't know what the reason might be. Several things come to mind: The data uncertainties are too small, e.g. you did not set uncertainties when you loaded the data and they defaulted to 1; your model does not fit well and you should use a different model; there is some data that's way off (e.g. your input table has -999 in there for invalid values). I would recommend to plot your fit and see how it looks. You'll probably be able to see by eye if  uncertainties and model are right.

**Comment by cosimoNigro** [comment]
@dtnguyen2 thanks, sorry for missing that part of the docs.

@hamogu thanks for your detailed explanation. Of course I checked the model / data agreement and properly included the errors. 
I am fitting a radiative model to the MWL SED of a blazar, and I have plenty of of X-ray and UV data with small uncertainties that make the final statistics very large. I added systematic errors on the flux and it helped a lot, giving me the 5.44 reduced statistics. I don't know what to do more than adding systematics, to improve the fit. Reducing the number of fitted parameters does not help either as it is difficult to establish a fixed value for some of them
![best_fit](https://user-images.githubusercontent.com/26409711/123083035-f0905e80-d41f-11eb-962e-c989ba5b07e3.png)

But my analysis has nothing to do with the issue, that was correctly addressed, so I am closing it.
Thanks
